src/stactools/noaa_hrrr/constants.py

```diff
@@ -0,0 +1,221 @@
+
+T = TypeVar("T", bound="StrEnum")
+
+ITEM_ID_FORMAT = "hrrr-{region}-{reference_datetime}-FH{forecast_hour}"
+COLLECTION_ID = "noaa-hrrr"
+
+EXTENDED_FORECAST_MAX_HOUR = 48
+STANDARD_FORECAST_MAX_HOUR = 18
+
+
+class StrEnum(str, Enum):
+    """A string-based enum, that can lookup an enum value from a string.
+
+    This is built-in in Python 3.11 but if you're not there yet...
+    """
+
+    @classmethod
+    def from_str(cls: Type[T], s: str) -> T:
+        """Look up an enum value by string."""
+        for value in cls:
+            if value == s:
+                return value
+        raise ValueError(f"Could not parse value from string: {s}")
+
+
+class CloudProvider(StrEnum):
+    """Cloud storage provider sources"""
+
+    azure = "azure"
+    aws = "aws"
+    google = "google"
+
+
+class Region(StrEnum):
+    """Values for the 'region' parameter in the HRRR hrefs"""
+
+    conus = "conus"
+    # Herbie can't find Alaska data as of 2024-05-30
+    alaska = "alaska"
+
+
+class Variable(StrEnum):
+    """Values for the 'variable' parameter in the HRRR hrefs"""
+
+    pressure = "prs"
+    native = "nat"
+    surface = "sfc"
+    sub_hourly = "subh"
+
+
+@dataclass
+class ForecastCycleType:
+    """Forecast cycle types"""
+
+    type: str
+
+    def __post_init__(self) -> None:
+        if self.type not in ["standard", "extended"]:
+            raise ValueError("Invalid forecast cycle type")
+
+        self.max_forecast_hour = (
+            STANDARD_FORECAST_MAX_HOUR
+            if self.type == "standard"
+            else EXTENDED_FORECAST_MAX_HOUR
+        )
+
+        # the available variables vary by forecast cycle type
+        self.variables = [
+            Variable.pressure,
+            Variable.native,
+            Variable.surface,
+        ]
+
+        if self.type == "standard":
+            self.variables.append(Variable.sub_hourly)
+
+    @classmethod
+    def from_timestamp_and_region(
+        cls, reference_datetime: datetime, region: Region
+    ) -> "ForecastCycleType":
+        """Determine the forecast cycle type based on the timestamp of the cycle run
+        hour
+
+        Extended forecasts are generated every six hours starting at hour 00
+        """
+        # no extended forecasts for alaska
+        if region == Region.alaska:
+            return cls("standard")
+
+        extended = reference_datetime.hour % 6 == 0
+        return cls("extended" if extended else "standard")
+
+    def generate_forecast_hours(self) -> List[int]:
+        """Generate a list of forecast hours for the given forecast cycle type"""
+        return list(range(1, self.max_forecast_hour + 1))
+
+    def validate_forecast_hour(self, forecast_hour: int) -> None:
+        """Check if forecast hour is valid for the forecast type.
+
+        Standard forecast cycles allow 0-18
+        Extended forecast cycles allow 0-48
+        """
+        valid = 0 <= forecast_hour <= self.max_forecast_hour
```

Comment:
I don't know if I would have thought to validate this to begin with but since this exists should it validate the actual hour is part of the set of possible hours which I think is (00, 06, 12, 18) ?